### PR TITLE
Doc page updates (inspecting circuits + autograph guide)

### PR DESCRIPTION
**Pull request by isaacdevlugt**
**Context:** Clarifications added to the autograph guide and the inspecting circuits section in sharp bits.

**Description of the Change:**

**Benefits:** 

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

File: doc/dev/autograph.rst [hunk 1]
```diff
@@ -622,6 +639,11 @@ Array(8.4, dtype=float64)
 Temporary variables used inside a loop --- and that are **not** passed to a
 function within the loop --- do not have any type restrictions.
 
+Python list manipulations
+-------------------------
+
+
+
```
Comment (isaacdevlugt) [inline doc/dev/autograph.rst:646]:
```suggestion
```

File: doc/dev/autograph.rst [hunk 2]
```diff
@@ -464,6 +463,24 @@ for instance by indexing a Python list with it. In that case, the list should be
...
+...
+>>> f()
+UserWarning: Tracing of an AutoGraph converted for loop failed with an exception:
+AutoGraphError: The variable 'my_list' was initialized with type <class 'list'>, which is not compatible with JAX. Typically, this is the case for
+non-numeric values. You may still use such a variable as a constant inside a loop, but it cannot be updated from one iteration to the next, 
+or accessed outside the loop scope if it was defined inside of it.
+...
+[Array(0, dtype=int64), Array(1, dtype=int64)]
+
+In this case, the code still executes, but AutoGraph is telling us that it fell back to executing the loop at compile time.
```
Comment (dime10) [inline doc/dev/autograph.rst:483]: Thank you Isaac! This is actually a good opportunity to document the currently supported pattern for accumulating results (with the caveat that the number and type of values must be known, and that they are homogenous):

```py
>>> @qjit(autograph=True)
... def f():
...     my_list = jnp.empty(2, dtype=int)
...     for i in range(2):
...         my_list[i] = i
...     return my_list
...
>>> f()
```

**Review by dime10 — APPROVED** [review]
Nice!

File: doc/dev/autograph.rst (hunk 2 above)
Comment (isaacdevlugt) [reply doc/dev/autograph.rst:483]: Good suggestion! How about this:

```suggestion
In this case, the code still executes, but AutoGraph is telling us that it fell back to executing the loop at compile time. Instead, results can be accumulated by indexing into an empty, but fixed in shape/size, JAX array:

>>> @qjit(autograph=True)
... def f():
...     my_list = jnp.empty(2, dtype=int)
...     for i in range(2):
...         my_list[i] = i
...     return my_list
...
>>> f()
Array([0, 1], dtype=int64)

Here, AutoGraph is able to properly capture the results.
```

Comment (dime10) [reply doc/dev/autograph.rst:483]: Perfect, I made two wording changes to the suggestion but it looks great 👍

**Review by josh146 — APPROVED** [review]
a nice addition :)

**Pull request merged**